new-lamassu-admin/src/pages/Analytics/components/wrappers/HourOfDayWrapper.js

```diff
@@ -0,0 +1,146 @@
+import { Box } from '@material-ui/core'
+import { makeStyles } from '@material-ui/core/styles'
+import moment from 'moment'
+import * as R from 'ramda'
+import React, { useState } from 'react'
+
+import { RadioGroup, Select } from 'src/components/inputs'
+import { H2 } from 'src/components/typography'
+import { MINUTE } from 'src/utils/time'
+
+import styles from '../../Analytics.styles'
+import Graph from '../../graphs/Graph'
+import LegendEntry from '../LegendEntry'
+
+const useStyles = makeStyles(styles)
+
+const options = [
+  { code: 'hourOfDayTransactions', display: 'Transactions' },
+  { code: 'hourOfDayVolume', display: 'Volume' }
+]
+
+const dayOptions = R.map(
+  it => ({
+    code: R.toLower(it),
+    display: it
+  }),
+  moment.weekdays()
+)
+
+const HourOfDayBarGraphHeader = ({
+  title,
+  representing,
+  period,
+  data,
+  machines,
+  selectedMachine,
+  handleMachineChange,
+  timezone,
+  currency
+}) => {
+  const classes = useStyles()
+
+  const [graphType, setGraphType] = useState(options[0].code)
+  const [selectedDay, setSelectedDay] = useState(dayOptions[0])
+
+  const legend = {
+    cashIn: <div className={classes.cashInIcon}></div>,
+    cashOut: <div className={classes.cashOutIcon}></div>
+  }
+
+  const offset = parseInt(timezone.split(':')[1]) * MINUTE
+
+  const txsPerWeekday = R.reduce(
+    (acc, value) => {
+      const created = new Date(value.created)
+      // console.log('before', R.clone(created))
+      created.setTime(
+        created.getTime() + created.getTimezoneOffset() * MINUTE + offset
+      )
+      // console.log('after', R.clone(created))
+      switch (created.getDay()) {
+        case 0:
+          acc.sunday.push(value)
+          break
+        case 1:
+          acc.monday.push(value)
+          break
+        case 2:
+          acc.tuesday.push(value)
+          break
+        case 3:
+          acc.wednesday.push(value)
+          break
+        case 4:
+          acc.thursday.push(value)
+          break
+        case 5:
+          acc.friday.push(value)
+          break
+        case 6:
+          acc.saturday.push(value)
+          break
+        default:
+          throw new Error('Day of week not recognized')
+      }
+      return acc
+    },
+    R.reduce(
+      (acc, value) => {
+        acc[value] = []
+        return acc
+      },
+      {},
+      R.map(it => it.code, dayOptions)
+    ),
```

Comment:
If I understand this correctly, this could be written as

```suggestion
R.zipObj(R.map(it => it.code, dayOptions), R.repeat([], 7))
```

or even better

```suggestion
R.fromPairs(R.map(it => [it.code, []], dayOptions))
```